File: A2/A2-014/A2-014.cpp
#include<iostream>
using namespace std;

int talismanSize, size01, size02, wCount = 0, wConsec = 0, wMaxConsec = 0;
string name01, name02, talisman;

/**
 * @brief Check if letter a or b contains 'l', 'o', 'v' or 'e'
 *
 * @param a is a first letter
 * @param b is a second letter
 * @return true if the letter contains 'l', 'o', 'v' or 'e'
 * @return false if the letter contains 'l', 'o', 'v' or 'e'
 */
bool isLove(char &a, char &b)
{
    bool c1 = (a == 'l' || b == 'l' || a == 'L' || b == 'L');
    bool c2 = (a == 'o' || b == 'o' || a == 'O' || b == 'O');
    bool c3 = (a == 'v' || b == 'v' || a == 'V' || b == 'V');
    bool c4 = (a == 'e' || b == 'e' || a == 'E' || b == 'E');
    return c1 || c2 || c3 || c4;
}

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input names
    cin >> name01 >> name02;
    size01 = name01.size();
    size02 = name02.size();

    // Add '$' and 'w' to a talisman
    talismanSize = max(size01, size02);
    for(int i = 0; i < talismanSize; i++)
    {
        // Add 'w' when a letter contains 'l', 'o', 'v' or 'e'
        if(isLove(name01[i % size01], name02[i % size02]))
        {
            talisman += 'w';
            // Count amount of 'w' and consecutive 'w'
            wCount++;
            wConsec++;
        }
        // Otherwise, add '$'
        else
        { 
            talisman += '$';
            // Find maximum consecutive 'w' and reset consecutive counter
            wMaxConsec = max(wMaxConsec, wConsec);
            wConsec = 0;
        }
    }
    // Don't forget to find maximum consecutive 'w' once again
    wMaxConsec = max(wMaxConsec, wConsec);

    // Add maximum consecutive 'w' at the end, if w amount is odd
    if(wCount % 2 == 1)
    {
        talisman += to_string(wMaxConsec);
    }
    // Add '#' if maximum consecutive is less than 2 and w amount is even
    else if(wMaxConsec < 2) { talisman += '#'; }

    // Output
    cout << talisman;
    return 0;
}

File: A2/A2-014/A2-014.py
# Input names and lowercase them
name01 = input().strip().lower()
name02 = input().strip().lower()

# Initialize a talisman
talisman = ""
talisman_size = max(len(name01), len(name02))
w_count = 0
w_consecutive = 0
w_max_consecutive = 0

# Add '$' and 'w' to a talisman
for i in range(talisman_size):
    # Add 'w' when a letter contains 'l', 'o', 'v' or 'e'
    c1 = name01[i % len(name01)] in ['l', 'o', 'v', 'e']
    c2 = name02[i % len(name02)] in ['l', 'o', 'v', 'e']
    if(c1 or c2):
        talisman += 'w'
        # Count amount of 'w' and consecutive 'w'
        w_count += 1
        w_consecutive += 1
    # Otherwise, add '$'
    else:
        talisman += '$'
        # Find maximum consecutive 'w' and reset consecutive counter
        w_max_consecutive = max(w_max_consecutive, w_consecutive)
        w_consecutive = 0
# Don't forget to find maximum consecutive 'w' once again
w_max_consecutive = max(w_max_consecutive, w_consecutive)

# Add maximum consecutive 'w' at the end, if w amount is odd
if(w_count % 2 == 1):
    talisman += str(w_max_consecutive)
# Add '#' if maximum consecutive is less than 2 and w amount is even
elif(w_max_consecutive < 2):
    talisman += '#'

# Output
print(talisman)